### PR TITLE
Resolved segfault while running CTS for nnhal

### PR DESCRIPTION
**Pull request by rairatne**
This includes disabling operations and adding validation checks for operations
causing CTS crashes. For further details please refer to individual commits

### CODE REVIEW

File: ngraph_creator/operations/src/Add.cpp [hunk 1]
```diff
@@ -12,14 +12,39 @@ Add::Add(int operationIndex) : OperationsBase(operationIndex) {
...
+    auto operandIndex1 = sModelInfo->getOperationInput(mNnapiOperationIndex, 0);
+    auto operandIndex2 = sModelInfo->getOperationInput(mNnapiOperationIndex, 1);
+    const auto& elementType1 = sModelInfo->getOperandType(operandIndex1);
+    const auto& elementType2 = sModelInfo->getOperandType(operandIndex2);
+    if ( !isValidInputTensor(0) || !isValidInputTensor(1) ) {
+         ALOGE("%s Empty  or Invalid dimensions size for input", __func__);
+         return false;
+    }
+    //check operand lifetime
+    const auto& dimsOperandIndex1 = sModelInfo->getOperationInput(mNnapiOperationIndex, 0);
```
Comment (akodanka) [inline ngraph_creator/operations/src/Add.cpp:24]: Isn't this same as operandIndex1 ?

**Review by akodanka — COMMENTED** [review]
In general, in all the places where we have put a check for Const-lifetime inputs, those operations might report as not supported for most tests, right ?
Do you know what has changed from v1.2 to v1.3 that would be causing the crash for non-Const lifetime inputs ?
Coz, Even the input nodes to a graph would be non-const.

File: ngraph_creator/operations/src/Squeeze.cpp [hunk 2]
```diff
@@ -12,17 +12,32 @@ Squeeze::Squeeze(int operationIndex) : OperationsBase(operationIndex) {
...
-    // The empty 2nd argument in Squeeze op causes dynamic output
-    // To add support, the dims will have to be calculated statically
-    if (sModelInfo->isOmittedInput(mNnapiOperationIndex, 1) ||
-        !sModelInfo->isOperandLifeTimeConst(dimsOperandIndex)) {
+    if (!sModelInfo->isOperandLifeTimeConst(dimsOperandIndex1)) {
         ALOGE("%s Only Constant dimensions supported now", __func__);
         return false;
     }
+    const auto& inputsSize = sModelInfo->getOperationInputsSize(mNnapiOperationIndex);
+    if (inputsSize == 2) {
```
[the remark this answers is not in the document]
Comment (rairatne) [reply ngraph_creator/operations/src/Squeeze.cpp:30]: It's check for optional parameter, if the optional parameter is not there then also its valid case!!

File: ngraph_creator/operations/src/Add.cpp [hunk 3]
```diff
@@ -12,14 +12,39 @@ Add::Add(int operationIndex) : OperationsBase(operationIndex) {
...
+    const auto& dimsOperandIndex2 = sModelInfo->getOperationInput(mNnapiOperationIndex, 1);
+    const auto& dimsOperandIndex3 = sModelInfo->getOperationInput(mNnapiOperationIndex, 2);
+    if(!sModelInfo->isOperandLifeTimeConst(dimsOperandIndex1) ||
+        !sModelInfo->isOperandLifeTimeConst(dimsOperandIndex2) ||
+        !sModelInfo->isOperandLifeTimeConst(dimsOperandIndex3)) {
+        ALOGE("%s Only Const lifetime is supported", __func__);
```
[the remark this answers is not in the document]
Comment (rairatne) [reply ngraph_creator/operations/src/Add.cpp:28]: Yes, issue is with mkldnn plugin, earlier there were no randomgraphgen test having lifetime as subgraphInput, So this change is kind of a workaround for now!!

**Review by akodanka — APPROVED** [review]
WA fixes for v1.3 NNHAL.
LGTM.

**Review by JeevakaPrabu — APPROVED** [review]

**Review by JeevakaPrabu — APPROVED** [review]

**Pull request merged**